### DISCUSSION TITLE
method chaining options in datatable

### DISCUSSION THREAD

**Discussion by samukweku** [h2oai/datatable#2863] [Ideas] [answered]
Just a thought on having the functions in datatable also available as methods; primary interest is for visual aspects of it:

``DT[:, dt.rowsum(dt.sum(f[:]))]`` 

compared to 

``DT[:, f[:].rowsum().sum()]``

silly example; hopefully it illustrates my thoughts on the visual aspect of it, and how it may help with readable code.

Presently , I abstract code to variables and pass the variables to the frame's brackets:

``variable = dt.rowsum(dt.sum(f[:]))``

``
DT[:, variable]
``

**Comment by st-pasha** [comment, marked as answer]
I like this idea, the syntax looks clean. Though it will increase the number of things we need to document.

**Reply by samukweku** [reply]
Great. I'm sure we can incrementally work on documentation
